sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -567,6 +567,7 @@ object DataSource extends Logging {
     val parquet = classOf[ParquetFileFormat].getCanonicalName
     val csv = classOf[CSVFileFormat].getCanonicalName
     val libsvm = "org.apache.spark.ml.source.libsvm.LibSVMFileFormat"
+    val image = "org.apache.spark.ml.source.image.ImageFileFormat"
```

Comment:
Why is this needed?

Reply:
We did it for libsvm for historical reasons. Since image is a new source, I don't think we have compatibility issues.